stan/math/prim/mat/prob/ordered_logistic_lpmf.hpp

```diff
@@ -73,25 +81,101 @@ namespace stan {
-        return -log1p_exp(lambda - c(0));
+        if (y == 1) {
+          logp_n += -log1p_exp(lambda - c[0]);
+        } else if (y == K) {
+          logp_n += -log1p_exp(c[K-2] - lambda);
+        } else {
+          logp_n += log_inv_logit_diff(c[y-2] - lambda,
+                                    c[y-1] - lambda);
+        }
+      return logp_n;
+    }
+
+    template <typename T_lambda, typename T_cut>
+    typename boost::math::tools::promote_args<T_lambda, T_cut>::type
+    ordered_logistic_lpmf(int y,
+                      const T_lambda& lambda,
+                      const Eigen::Matrix<T_cut, Eigen::Dynamic, 1>& c)  {
+      return ordered_logistic_lpmf<false>(y, lambda, c);
+    }
+
+    /**
+     * Returns the (natural) log probability of the specified array 
+     * of integers given the vector of continuous locations and 
+     * specified cutpoints in an ordered logistic model.
+     *
+     * <p>Typically the continous location
+     * will be the dot product of a vector of regression coefficients
+     * and a vector of predictors for the outcome.
+     *
+     * @tparam propto True if calculating up to a proportion.
+     * @tparam T_lambda Location type.
+     * @tparam T_cut Cut-point type.
+     * @param y Array of integers
+     * @param lambda Vector of continuous location variables.
+     * @param c Positive increasing vector of cutpoints.
+     * @return Log probability of outcome given location and
+     * cutpoints.
+
+     * @throw std::domain_error If the outcome is not between 1 and
+     * the number of cutpoints plus 2; if the cutpoint vector is
+     * empty; if the cutpoint vector contains a non-positive,
+     * non-finite value; or if the cutpoint vector is not sorted in
+     * ascending order.
+     * @throw std::invalid_argument If y and lambda are different
+     * lengths.
+     */
+    template <bool propto, typename T_lambda, typename T_cut>
+    typename boost::math::tools::promote_args<T_lambda, T_cut>::type
+    ordered_logistic_lpmf(const std::vector<int>& y,
+                      const Eigen::Matrix<T_lambda, Eigen::Dynamic, 1>& lambda,
+                      const Eigen::Matrix<T_cut, Eigen::Dynamic, 1>& c) {
+      typename boost::math::tools::promote_args<T_lambda, T_cut>::type
+        logp_n(0.0);
 
-      // log(inv_logit(lambda - c(K-3)));
-      if (y == K) {
-        return -log1p_exp(c(K-2) - lambda);
+      using boost::math::tools::promote_args;
+      using std::exp;
+      using std::log;
+
+      static const std::string function = "ordered_logistic";
+
+      int N = lambda.size();
+      int K = c.size() + 1;
+
+      check_consistent_sizes(function, "Integers", y, "Locations", lambda);
+
+      for (int i = 0; i < N; ++i) {
+      check_bounded(function, "Random variable", y[i], 1, K);
+      check_finite(function, "Location parameter", lambda[i]);
       }
+      for (int i = 1; i < c.size(); ++i)
+        check_greater(function, "Cut points parameter", c(i), c(i - 1));
```

Comment:
Is there no `check_ordered`?  If not, there should be, as this won't produce a good message because `i` will be lost.